### DISCUSSION TITLE
Error removing DRM from Nook Books

### DISCUSSION THREAD

**Discussion by iamcaptainskye** [apprenticeharper/DeDRM_tools#2209] [Q&A] [answered]
Hey all,

Tried to use the FAQ, but I'm not finding my specific issue and my tech literacy is limited. I'm using Calibre 5.5 and DeDRM 7.2.1, on Windows 10 (version 21H2). I've got the Nook books downloaded - I don't have Adobe on my computer (all my PDFs open in Firefox) so I don't think I have an issue of the Adobe PDF's being saved in the new format. I've included the Calibre log below so hopefully someone can point me in the right direction. I'm eventually hoping to get these on Kindle, but I've used Calibre before and know how to do everything once the DRM is gone.

Please let me know if there's other info that would be helpful to know.

calibre Debug log
calibre 5.5 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (7, 2, 1) && ePub Extended Metadata {Writer} (0, 8, 1) && ePub Extended Metadata {Reader} (0, 8, 1) && ePub Extended Metadata (0, 8, 1)
calibre 5.5 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (7, 2, 1) && ePub Extended Metadata {Writer} (0, 8, 1) && ePub Extended Metadata {Reader} (0, 8, 1) && ePub Extended Metadata (0, 8, 1)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 120.0 x 120.0
physicalDpi: 157.8252427184466 x 157.65517241379308
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.55] splash screen shown
[0.55] Initializing db...
[0.55] db initialized
[0.55] Constructing main UI...
[1.59] main UI initialized...
[1.59] Hiding splash screen
Starting QuickView
DeDRM v7.2.1: Trying to decrypt 2940152130034.epub
DeDRM v7.2.1: Verifying zip archive integrity
DeDRM v7.2.1: “2940152130034.epub” is a secure Barnes & Noble ePub
DeDRM v7.2.1: Looking for new NOOK Study Keys after 0.2 seconds
No nook Study log files have been found.
DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 644, in run
  File "calibre_plugins.dedrm.__init__", line 313, in ePubDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
[7.64] splash screen hidden
[7.64] Started up in 7.64 seconds with 0 books
Added Grimm's Fairy Stories to db in: 0.1
Added 1 books in 0.9 seconds
DeDRM v7.2.1: Trying to decrypt 2940152130041.epub
DeDRM v7.2.1: Verifying zip archive integrity
DeDRM v7.2.1: “2940152130041.epub” is a secure Barnes & Noble ePub
DeDRM v7.2.1: Looking for new NOOK Study Keys after 0.1 seconds
No nook Study log files have been found.
DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 644, in run
  File "calibre_plugins.dedrm.__init__", line 313, in ePubDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md

**Comment by ElleKayEm** [comment, marked as answer]
Need to get your B&N key.  DeDRM 10.0.3 has a way to extract it from the Windows Nook app.

Get it here: https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.3

Then in DeDRM customization, click ADE PassHash (B&N) ebooks and then the green plus sign.  Select Extract key from Nook Windows application.  Give it whatever name you want.  Click ok and apply on your way out.  Then restart calibre before re-adding your books.  The Nook app needs to be on the same machine and user as calibre.  And signed into your account of course.

**Reply by iamcaptainskye** [reply]
You are brilliant and a treasure and a genius. Thank you so much!!!!

**Comment by gunday** [comment]
<img width="494" alt="Screenshot 2024-10-21 at 12 15 41" src="https://github.com/user-attachments/assets/4274e48b-c1f9-4de2-b8c4-681fa15f3993">
My shows this msg. Im using mac os. The is any idea how can I fix it?

**Reply by ElleKayEm** [reply]
Need to extract your key from somewhere first.  Not sure if it's currently possible any more.  The Windows Nook app is no longer available.  The old Mac apps were disabled long before that.  People used to get it out of an old version of the Android app using an emulator or a rooted Android or Nook device.  But I don't know if that's possible now either.  How are you downloading the books?

**Reply by gunday** [reply]
I’m using Android Studio(emulate android 9) to download the books to my virtual Android, then sending a copy to the virtual SD card and uploading it to 4shared to download on my Mac. However, when I try to open the PDF, it keeps asking for a password. So I burned dollars in books thats is useless for me now. sad....

**Reply by ElleKayEm** [reply]
Not sure if decryption of B&N PDFs works either.  For a long time it was just their epubs.  Someone provided a script for their PDFs later which noDRM added to the plugin.  But I don't know if it was ever tested.
